source/app.d

```diff
@@ -50,18 +50,19 @@ class ImageFitness
     {
         foreach(ref circle; circles)
         {
-            auto color = circle.color;
-            foreach(x; (circle.x - circle.radius / 2) .. (circle.x + circle.radius / 2))
+            import std.algorithm.comparison: min, max;
+            auto fromX = max(circle.x - circle.radius, 0);
```

Comment:
Co mi to jenom...připomíná... https://github.com/Hnatekmar/GeneticAlgorithm/blob/fast4shoot/source/app.d#L112